### PR TITLE
Updated proj to 6.1.0

### PR DESCRIPTION
**Pull request by o1lo01ol1o**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -12644,8 +12644,12 @@ in
   primesieve = callPackage ../development/libraries/science/math/primesieve { };
 
   prison = callPackage ../development/libraries/prison { };
+  
+  proj = proj_6_1_0;
 
-  proj = callPackage ../development/libraries/proj { };
+  proj_6_1_0 = callPackage ../development/libraries/proj { };
+
+  proj_5_2_0 = callPackage ../development/libraries/proj/5.2.0.nix { };
```
Comment (vbgl) [inline pkgs/top-level/all-packages.nix]: Why is version 5.2.0 kept?

**Review by vbgl — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (o1lo01ol1o) [reply pkgs/top-level/all-packages.nix]: Because I misread [this](https://nixos.org/nixpkgs/manual/#sec-organisation).  Happy to take it out if you like.

Comment (o1lo01ol1o) [reply pkgs/top-level/all-packages.nix]: @vbgl done.

**Comment by vbgl** [comment]
@GrahamcOfBorg build libspatialite

**Comment by vbgl** [comment]
This breaks libspatialite; here is a possible patch:

~~~
diff --git a/pkgs/development/libraries/libspatialite/default.nix b/pkgs/development/libraries/libspatialite/default.nix
index 8e7b539527e..df8134ef9e0 100644
--- a/pkgs/development/libraries/libspatialite/default.nix
+++ b/pkgs/development/libraries/libspatialite/default.nix
@@ -17,6 +17,8 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
+  CFLAGS = "-DACCEPT_USE_OF_DEPRECATED_PROJ_API_H=1";
+
   postInstall = "" + optionalString stdenv.isDarwin ''
     ln -s $out/lib/mod_spatialite.{so,dylib}
   '';
~~~

**Comment by o1lo01ol1o** [comment]
Would you prefer that or linking libspatialite to the previous version? I believe the latter keeps the bounds closer to the documented bounds for libspatialite.

**Comment by o1lo01ol1o** [comment]
@GrahamcOfBorg build libspatialite

**Comment by o1lo01ol1o** [comment]
@vbgl I'm not sure I'm authorized to tell 

@GrahamcOfBorg build libspatialite

but this builds for me locally

**Comment by Mic92** [comment]
@GrahamcOfBorg build libspatialite

File: pkgs/development/libraries/proj/default.nix [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
   src = fetchurl {
-    url = https://download.osgeo.org/proj/proj-5.2.0.tar.gz;
-    sha256 = "0q3ydh2j8qhwlxmnac72pg69rw2znbi5b6k5wama8qmwzycr94gg";
+    url = https://download.osgeo.org/proj/proj-6.1.0.tar.gz;
+    sha256 = "0240g1jdqj6qj40sjrz8bwi0nhzb8irvv71llhyz1lhr8g2naqb7";
   };
 
   doCheck = stdenv.is64bit;
 
+  buildInputs = [ sqlite pkg-config ];
```
Comment (Mic92) [inline pkgs/development/libraries/proj/default.nix]: pkg-config belongs to `nativeBuildInputs`.

**Comment by o1lo01ol1o** [comment]
@vbgl ping

**Comment by vbgl** [comment]
@GrahamcOfBorg build python37Packages.cartopy

**Comment by vbgl** [comment]
You also need to update `libgeotiff`. Here is a possible patch.

~~~
diff --git a/pkgs/development/libraries/libgeotiff/default.nix b/pkgs/development/libraries/libgeotiff/default.nix
index 33b9e7830e7..abdaa69ba59 100644
--- a/pkgs/development/libraries/libgeotiff/default.nix
+++ b/pkgs/development/libraries/libgeotiff/default.nix
@@ -1,21 +1,20 @@
-{ stdenv, fetchurl, libtiff, libjpeg, proj, zlib}:
+{ stdenv, fetchurl, libtiff, libjpeg, proj, zlib }:
 
 stdenv.mkDerivation rec {
-  version = "1.4.3";
+  version = "1.5.1";
   name = "libgeotiff-${version}";
 
   src = fetchurl {
-    url = "https://download.osgeo.org/geotiff/libgeotiff/${name}.tar.gz";
-    sha256 = "0rbjqixi4c8yz19larlzq6jda0px2gpmpp9c52cyhplbjsdhsldq";
+    url = "https://github.com/OSGeo/libgeotiff/releases/download/${version}/${name}.tar.gz";
+    sha256 = "0b31mlzcv5b1y7jdvb7p0pa3xradrg3x5g32ym911lbhq4rrgsgr";
   };
 
   configureFlags = [
     "--with-jpeg=${libjpeg.dev}"
     "--with-zlib=${zlib.dev}"
   ];
-  buildInputs = [ libtiff proj ];
 
-  hardeningDisable = [ "format" ];
+  buildInputs = [ libtiff proj ];
 
   meta = {
     description = "Library implementing attempt to create a tiff based interchange format for georeferenced raster imagery";
~~~

**Comment by o1lo01ol1o** [comment]
@vbgl building for me.

**Comment by vbgl** [comment]
@GrahamcOfBorg build python37Packages.cartopy

**Comment by vbgl** [comment]
This breaks `python3Packages.cartopy`. Here is the build error:

~~~
In file included from lib/cartopy/_trace.h:27:0,
                 from lib/cartopy/trace.cpp:608:
/nix/store/vwkxzx4iwwjk10rmq7i2c7phr9mq8svj-proj-6.1.0/include/proj_api.h:37:2: error: #error 'To use the proj_api.h you must define the macro ACCEPT_USE_OF_DEPRECATED_PROJ_API_H'
 #error 'To use the proj_api.h you must define the macro ACCEPT_USE_OF_DEPRECATED_PROJ_API_H'
  ^~~~~
/nix/store/8l17hs2nyc79432hw3caw2vzx5z00d78-python3.7-Cython-0.29.7/lib/python3.7/site-packages/Cython/Compiler/Main.py:367: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /build/Cartopy-0.17.0/lib/cartopy/trace.pyx
  tree = Parsing.p_module(s, pxd, full_module_name)
error: command 'gcc' failed with exit status 1
~~~

This breaks `saga_2_3_2`:

~~~
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../../../.. -fPIC -I../../../../src/saga_core -D_SAGA_LINUX -D_TYPEDEF_BYTE -D_TYPEDEF_WORD -D_SAGA_UNICODE -fopenmp -lgomp -c crs_assign.cpp  -fPIC -DPIC -o .libs/crs_assign.o
In file included from PROJ4_Shapes.h:68:0,
                 from MLB_Interface.cpp:104:
PROJ4_Base.h:71:10: fatal error: projects.h: No such file or directory
 #include <projects.h>
          ^~~~~~~~~~~~
compilation terminated.
make[5]: *** [Makefile:562: MLB_Interface.lo] Error 1
~~~

**Comment by FRidh** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/64551

**Pull request closed without merging**